src/pages/patch/BuildVariants.tsx

```diff
@@ -35,27 +45,86 @@ export const BuildVariants: React.FC = () => {
         <Skeleton active title={false} paragraph={{ rows: 4 }} />
       )}
       {data &&
-        data.patchBuildVariants.map(({ displayName, tasks }) => (
+        data.patchBuildVariants.map(({ displayName, tasks, variant }) => (
           <BuildVariant
             key={`buildVariant_${displayName}`}
-            className="patch-build-variant"
+            data-cy="patch-build-variant"
           >
-            <P1>{displayName}</P1>
-            <VariantTasks>
-              {tasks.map((task) => (
-                <TaskSquare key={task.id} {...task} />
-              ))}
-            </VariantTasks>
+            <P1>
+              <Link
+                to={`${getVersionRoute(id)}?page=0&variant=${variant}`}
+                onClick={() =>
+                  patchAnalytics.sendEvent({
+                    name: "Click Build Variant Grid Link",
+                  })
+                }
+              >
+                {displayName}
+              </Link>
+            </P1>
+            <VariantTaskGroup variant={variant} tasks={tasks} />
           </BuildVariant>
         ))}
     </SiderCard>
   );
 };
 
+interface VariantTaskGroupProps {
+  tasks: PatchBuildVariantTask[];
+  variant: string;
+}
+const VariantTaskGroup: React.FC<VariantTaskGroupProps> = ({
+  tasks,
+  variant,
+}) => {
+  const groupedTasks = groupTasksByColor(tasks);
+  return (
+    <VariantTasks>
+      {Object.keys(groupedTasks).map((color) => (
+        <GroupedTaskSquare
+          key={`${variant}_${color}`}
+          statuses={groupedTasks[color].statuses}
+          count={groupedTasks[color].count}
+          color={color}
+          textColor={groupedTasks[color].textColor}
+        />
+      ))}
+    </VariantTasks>
+  );
+};
+
 const BuildVariant = styled.div`
   margin-bottom: 8px;
 `;
 const VariantTasks = styled.div`
   display: flex;
   flex-wrap: wrap;
 `;
+
+const groupTasksByColor = (tasks: PatchBuildVariantTask[]) => {
+  const taskColors: {
+    [key: string]: { count: number; statuses: string[]; textColor: string };
+  } = {};
+  tasks.forEach((task) => {
+    if (taskColors[mapVariantTaskStatusToColor[task.status]]) {
```

Comment:
I would save mapVariantTaskStatusToColor[task.status] in a separate variable so that this is easier to read